libs/accounts/recovery-phone/src/lib/recovery-phone.service.ts

```
@@ -368,6 +338,46 @@ export class RecoveryPhoneService {
     return await this.recoveryPhoneManager.removePhoneNumber(uid);
   }
 
+  /**
+   * Updates the existing recovery phone number with the new one associated to the code provided.
+   *
+   * @param uid
+   * @param code
+   * @returns
+   */
+  public async replacePhoneNumber(uid: string, code: string): Promise<boolean> {
+    const unconfirmedCode = await this.recoveryPhoneManager.getUnconfirmed(
+      uid,
+      code
+    );
+    if (unconfirmedCode == null) {
+      // there are no codes for the new number, can't proceed.
+      return false;
+    }
+    const { phoneNumber, isSetup } = unconfirmedCode;
+
+    if (isSetup !== true) {
+      // this code was not for a setup operation, can't proceed.
+      return false;
+    }
+
+    // both of these have potential to throw RecoveryNumberNotSupportedError.
```

I'd love some thoughts here. I had these wrapped in a `try/catch` but it felt silly to catch specific errors just to throw them again. If these fail then there's not much to be done except let it bubble up

---

I dont think you need to add the extra try/catch.

---

Yep let them bubble up.